### PR TITLE
Adding ability to handle OverlayFS mounts

### LINKED ISSUE #2218: Lacking support for OverlayFS in the mount processing
When I try to start Kubernetes it fails with the following error:

failed to get device for dir "/var/lib/kubelet": could not find device with major: 0, minor: 22 in cached partitions map

The missing device is described in the following way in my mountinfo file (/proc/self/mountinfo):

29 15 0:22 / /var/lib rw,relatime shared:18 - overlay overlay rw,lowerdir=/var/lib,upperdir=/mnt/flash/var/lib,workdir=/mnt/flash/.work-var-lib

I need support for overlayfs in the processMounts function (fs/fs.go) to make Kubernetes work.

### PR DESCRIPTION
**Pull request by jaerik**
Fixes #2218
Signed-off-by: Erik Jansson <erikja@axis.com>

### CODE REVIEW

**Comment by jaerik** [comment]
email address updated

File: fs/fs.go [hunk 1]
```diff
@@ -222,6 +223,10 @@ func processMounts(mounts []*mount.Info, excludedMountpointPrefixes []string) ma
 			}
 		}
 
+		if mount.Fstype == "overlay" {
+			mount.Source = fmt.Sprintf("%s_%d-%d", mount.Source, mount.Major, mount.Minor)
```
Comment (dashpole) [inline fs/fs.go]: why are we setting source here?

Comment (jaerik) [reply fs/fs.go]: When mount.Source is not unique to overlayfs mount in the mountinfo file, the mount.Source is "overlay" for all overlayfs mounts, it causes problems when it is used as key in the partitions map. Hence this is a way to make mount.Source unique.

Comment (dashpole) [reply fs/fs.go]: can you add a comment explaining that this makes makes mount.Source unique for overlay mounts?

**Pull request merged**